net/Socket.hpp

```diff
@@ -152,40 +166,71 @@ class Socket
         // Doesn't block on sockets; no error handling needed.
 #if !MOBILEAPP
         ::close(_fd);
-        LOG_DBG("Closed socket to [" << clientAddress() << ']');
+        LOG_DBG("Closed socket " << toStringImpl());
 #else
         fakeSocketClose(_fd);
 #endif
     }
 
-    /// Create socket of the given type.
-    static int createSocket(Type type);
+    /// Returns true if this socket is open, i.e. allowed to be polled and not shutdown
+    bool isOpen() const noexcept { return _open; }
+    /// Returns true if this socket has been closed, i.e. rejected from polling and potentially shutdown
+    bool isClosed() const noexcept { return !_open; }
 
-    void setClientAddress(const std::string& address)
-    {
-        _clientAddress = address;
-    }
+    constexpr Type type() const noexcept { return _type; }
+    constexpr bool isIPType() const noexcept { return Type::IPv4 == _type || Type::IPv6 == _type; }
```

Comment:
I prefer qualification as much as possible, even though knowing a wrong `noexcept` is not ideal (-> disastrous termination in case of an exception). So yes, `noexcept` needs to be used carefully.
A `noexcept` gives benefits of relieving the compiler of creating a potential stack frame and is also required for `constexpr` semantics.

Now `constexpr` is not only useful for using an instance in a compile time manner (as I agree a Socket never will be), but also to
- force inline as much as possible
- force compiler to enhance code validation 
The latter is what interests me most, while the inlining is a very nice to have to allow compiler to remove complexity. This especially for small accessors. Allowing us to be more expressive at zero costs.